metricbeat/module/kibana/stats/stats.go

```diff
@@ -122,36 +121,39 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 // Fetch methods implements the data gathering and data conversion to the right format
 // It returns the event which is then forward to the output. In case of an error, a
 // descriptive error must be returned.
-func (m *MetricSet) Fetch(r mb.ReporterV2) {
+func (m *MetricSet) Fetch(r mb.ReporterV2) error {
 	now := time.Now()
 
-	m.fetchStats(r, now)
+	err := m.fetchStats(r, now)
+	if err != nil {
+		return err
+	}
+
 	if m.XPackEnabled {
 		m.fetchSettings(r, now)
 	}
+
+	return nil
 }
 
-func (m *MetricSet) fetchStats(r mb.ReporterV2, now time.Time) {
+func (m *MetricSet) fetchStats(r mb.ReporterV2, now time.Time) error {
 	content, err := m.statsHTTP.FetchContent()
 	if err != nil {
-		elastic.ReportAndLogError(err, r, m.Logger())
-		return
+		return err
 	}
 
 	if m.XPackEnabled {
 		intervalMs := m.calculateIntervalMs()
 		err = eventMappingStatsXPack(r, intervalMs, now, content)
 		if err != nil {
 			m.Logger().Error(err)
-			return
+			return nil
```

Comment:
Should this be return err here? Also is m.Logger above still needed?

Reply:
Please see the note in the PR description about x-pack code paths 😄.

Reply:
ahh good to know! Thanks!